File: frontend/lib/core/models/note.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'note.freezed.dart';
part 'note.g.dart';

@freezed
class Note with _$Note {
  const Note._();
  
  const factory Note({
    @JsonKey(name: '_id') required String id,
    String? userId,
    required String title,
    @JsonKey(name: 'body') required String content,
    @Default([]) List<String> tags,
    String? listId,
    @Default(false) bool isArchived,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _Note;

  factory Note.fromJson(Map<String, dynamic> json) => _$NoteFromJson(json);
}

File: frontend/src/lib/types/note.ts
// Note type definitions

export interface Note {
  _id: string;
  userId: string;
  listId?: string;
  title: string;
  body: string;
  tags: string[];
  isArchived: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateNoteData {
  title: string;
  body: string;
  tags?: string[];
  listId?: string;
}

export interface UpdateNoteData extends Partial<CreateNoteData> {
  isArchived?: boolean;
}

export interface NoteFilters {
  listId?: string;
  isArchived?: boolean;
  tags?: string[];
  search?: string;
}
